### PR TITLE
Bring additional next-article features and overrides to o-share

### PR DESCRIPTION
**Pull request by notlee**
**Note: this is not a breaking change but _will_ break the ft.com article page due to heavy overrides and customisation. I plan to release a beta, update next-article, then release for real.**

- Support icon labels
- Support custom share behavior (button with JS events or form submission)
- Remove margin except for the small variant to provide a reasonable touch target area
- Introduces a dependency on o-typography. This would usually be a breaking change but we can be confident that this is not a breaking change as they both depend on o-normalise v2 which was released at the same time (it's impossible to use any other major of o-typography with the current major of o-share)

before/after: default variant has no margin
Screenshot 2020-04-20 at 18 36 21
Screenshot 2020-04-20 at 18 36 24

before/after:  default variant with focus
Screenshot 2020-04-20 at 18 37 41
Screenshot 2020-04-20 at 18 37 44

before/after:  inverse variant

before/after:  vertical variant

before/after: small variant

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,29 +19,27 @@ Social media buttons.
 
 ## Markup
 
-The simplest markup you might need looks like this:
+The following markup is the simplest but requires JavaScript to work and does not support custom share icons or icon labels. If your project requires either of those features see the [full markup](#full-markup) seciton.
```
Comment (JakeChampion) [inline README.md:22]:
```suggestion
The following markup is the simplest but requires JavaScript to work and does not support custom share icons or icon labels. If your project requires either of those features see the [full markup](#full-markup) section.
```

**Comment by notlee** [comment]
Unsure why the fox failed here:
```
➜  o-share git:(label-support) npx @financial-times/origami-bundle-size-cli
npx: installed 116 in 18.734s
o-share bundle size difference from 7.3.1 to 5581f80ba11a6c941297f7a4e3c50c0d72cb0094
js: 0.85kb increase (0.32kb/gzip)
css: 6.25kb increase (0.42kb/gzip)
```

**Comment by notlee** [comment]
I neglected to give this a test in IE and windows high contrast mode.

Note to self before releasing a beta: 
- try outputting windows high contrast mode only once
- test in IE11
- test in windows high contrast mode

**Comment by notlee** [comment]
high contrast mode:
Screenshot 2020-04-20 at 19 55 26
Screenshot 2020-04-20 at 19 56 00

**Comment by origamiserviceuser** [comment]
👋 Percy has finished running the visual regression testing!

**Comment by origamiserviceuser** [comment]
👋 Percy has finished running the visual regression testing!

**Comment by notlee** [comment]
Percy is here! And look, I can see that existing icons of the small variant are unaffected in Chrome and Firefox (and soon other browsers)

Screenshot 2020-04-24 at 10 31 46

**Comment by origamiserviceuser** [comment]
:tada: This PR is included in version v7.4.0 :tada:

**Pull request merged**